ocp_resources/self_node_remediation_templates.py

```diff
@@ -0,0 +1,47 @@
+# API reference: https://github.com/medik8s/self-node-remediation
+from typing import Any, Union
+
+from ocp_resources.resource import NamespacedResource
+
+
+class SelfNodeRemediationTemplate(NamespacedResource):
+    """
+    SelfNodeRemediationTemplate CRD
+    """
+
+    api_group = NamespacedResource.ApiGroup.SELF_NODE_REMEDIATION_MEDIK8S_IO
+    VALID_STRATEGIES = {"Automatic", "ResourceDeletion", "OutOfServiceTaint"}
+
+    def __init__(
+        self,
+        remediation_strategy: Union[str, None] = "Automatic",
+        **kwargs: Any,
+    ) -> None:
+        """
+        Create SelfNodeRemediationTemplate object.
+
+        Args:
+            remediation_strategy (str, optional): The remediation method for unhealthy nodes.
+                Options are "Automatic", "OutOfServiceTaint", or "ResourceDeletion".
+                Default is "Automatic".
+        """
+        if remediation_strategy not in SelfNodeRemediationTemplate.VALID_STRATEGIES:
+            raise ValueError(
+                f"Invalid remediation strategy: {remediation_strategy}. "
+                f"Supported values are: {', '.join(SelfNodeRemediationTemplate.VALID_STRATEGIES)}"
+            )
+
+        super().__init__(
+            **kwargs,
+        )
+        self.remediation_strategy = remediation_strategy
+
+    def to_dict(self) -> None:
+        super().to_dict()
+        if not self.yaml_file:
+            spec = self.res.setdefault("spec", {})
+            spec["template"] = spec.get("template", {})
+            spec["template"]["spec"] = spec["template"].get("spec", {})
```

Comment:
please check how this is done in the examples referenced in the contribution doc